spinoso-string/src/lib.rs

```diff
@@ -1933,6 +1947,30 @@ impl String {
     /// assert_eq!(s.get_char_slice(Range { start: 15, end: 10 }), None);
     /// assert_eq!(s.get_char_slice(Range { start: 15, end:  1 }), None);
     /// assert_eq!(s.get_char_slice(Range { start:  4, end:  1 }), Some(&b""[..]));
+    ///
+    /// let s = String::ascii(b"abcd".to_vec());
```

Comment:
I add related tests to the doctest of `String#get_char_slice` before refactoring. If unit tests are preferred in this case, please let me know and I would be happy to refactor.

Reply:
I think since we've pushed the logic into the encoded variants, I'd prefer these to be unit tests in each of the encoded buf types.

Ideally the doc tests would only show examples of how to use the APIs. In addition to moving all of the doc tests to unit tests, do you want to take a stab at simplifying the examples to maybe show only one `Some` and `None` case for UTF-8 and ASCII?

Reply:
I have added unit tests for the three encoded variants, and then simplify the doc tests in  b4c8746c6d8ed4329a15036e52a8f0aee0e7ee78.

Reply:
thank you!